graph_wrap/django_rest_framework/api_transformer.py

```diff
@@ -86,7 +86,7 @@ def _build_graphene_type_name(self):
         else:
             named_field = self._serializer
         serializer_cls_name = self._serializer.__class__.__name__
-        if serializer_cls_name == 'NestedSerializer':
+        if 'Nested' in serializer_cls_name:
```

Comment:
I agree this is not a great check and I didn't particularly enjoy adding it. It's here for precisely the reason you mentioned - nothing came to mind as a way to be able to identify a NestedSerializer apart from the name. Especially difficult when the classes are dynamically created by DRF (otherwise we could have maybe relied on checking type).

My worry is that with this change we get the opposite problem - that too many serializers get deemed as nested ones. Could you perhaps describe your use case a bit more? I guess you have maybe overwritten the build_nested_field  method somewhere? https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/rest_framework/serializers.py

One option would be that there's some opitional config for graph wrap - something like NESTED_SERIALIZER_LOOK_UPS, which could be a list of strings the user wishes graph wrap to consider when deciding if a serializer is nested or not? Logic would be that if a class name is found which contains any one of the strings configured in NESTED_SERIALIZER_LOOK_UPS, it's considered a nested serializer. So for you, you'd have NESTED_SERIALIZER_LOOK_UPS=['Nested'] (and the default would be ['NestedSerializer']).

Reply:
In our case, we've taken the (perhaps uninformed) approach of explicitly declaring separate classes for nested serializers. So for a Django model named Foo, we have two DRF serializers: FooSerializer, with all fields, and NestedFooSerializer, which has only the bare minimum attributes like ID and name. (In retrospect, the naming format FooNestedSerializer probably would have been a better choice. I'm not opposed to renaming our serializers to better fit graph_wrap.)

What about checking for `.endswith('NestedSerializer')`? I'll dig into DRF a bit too to see if I can come up with anything more elegant. Thanks for the feedback!

Reply:
I think `.endswith('NestedSerializer')` would be a reasonable compromise if we can't figure out a way to identify a `NestedSerializer`. 

Mainly for my own reference, I think this is all we have to go by in terms of identifying features:

`
    def build_nested_field(self, field_name, relation_info, nested_depth):        """        Create nested fields for forward and reverse relationships.        """        class NestedSerializer(ModelSerializer):            class Meta:                model = relation_info.related_model                depth = nested_depth - 1                fields = '__all__'
        field_class = NestedSerializer        field_kwargs = get_nested_relation_kwargs(relation_info)
        return field_class, field_kwargs
`

Reply:
What about checking for the presence of `Meta.depth`? Seems like that would only be set in nested serializers.

Reply:
I came up with an alternate approach in #11; let me know what you think!

Reply:
> What about checking for the presence of `Meta.depth`? Seems like that would only be set in nested serializers.

Meta.depth is in general what _creates_ a NestedSerializer. It will often be present on a non-nested serializer: take the PostSerializer from the graph wrap test suite as an example (and refer to https://www.django-rest-framework.org/api-guide/serializers/#specifying-nested-serialization).

Reply:
I'm actually wondering if we're talking at cross-purposes. When I re-read your use case, I think you might be referring to a "nested serializer" as any serializer which is embedded in another serializer? Whereas I'm talking explicitly about the NestedSerializer built dynamically by this function when a depth arg is set:
```
    def build_nested_field(self, field_name, relation_info, nested_depth):
        """
        Create nested fields for forward and reverse relationships.
        """
        class NestedSerializer(ModelSerializer):
            class Meta:
                model = relation_info.related_model
                depth = nested_depth - 1
                fields = '__all__'

```

Reply:
> I'm actually wondering if we're talking at cross-purposes.

I probably just have no idea what I'm doing. :slightly_smiling_face: I've opened #12 to better document the issue I'm running into; please let me know if that makes sense.